src/main/java/dk/cngroup/lentils/controller/HintController.java

```diff
@@ -136,18 +136,22 @@ public String saveNewHint(@Valid @ModelAttribute(TEMPLATE_ATTR_HINT) final HintF
             return VIEW_ADMIN_HINT_DETAIL;
         }
         Hint hint = new Hint(cypherService.getCypher(cypherId));
-        imageService.saveImageFile(hintFormDto);
         mapper.map(hintFormDto, hint);
         hintService.save(hint);
+        if (FileUtils.isFilePresentInHintForm(hintFormDto)) {
+            imageService.saveImageFile(hintFormDto, hint.getImage().getPath());
+        }
         return REDIRECT_ADMIN_HINT_LIST + CYPHERID_PARAMETER + hint.getCypherId();
     }
 
-    private void addImageUri(final Model model, final HintFormDTO hintFormDto) {
+    private void addImageUri(final Model model, final HintFormDTO hintFormDto, final Long hintId) {
         if (hintFormDto.getImage() == null) {
             model.addAttribute(TEMPLATE_ATTR_FILE, "");
         } else {
             model.addAttribute(TEMPLATE_ATTR_FILE, MvcUriComponentsBuilder.fromMethodName(HintController.class,
-                    "serveFile", hintFormDto.getImage().getOriginalFilename()).build().toString());
-        }
+                    "serveFile", imageService.getPureFileName(
```

Comment:
Tuto sekci bych trosku rozdelil z formatoval, protoze je to hodne slozite na precteni.
Muzes to prosim zkusit malinko zrefaktorovat? Vytvorit nejake pomocne promenne atd.? :) Zkus tak, jak to citis ty a ja ti kdyztak poradim, jak bych na to sel treba jeste ja ;)